src/components/AddressInputPanel/index.tsx

```diff
@@ -173,24 +188,48 @@ export default function AddressInputPanel({
     <InputPanel>
       <ContainerRow error={input !== '' && error}>
         <InputContainer>
-          <InputRow>
+          <AutoColumn gap="md">
+            <RowBetween>
+              <TYPE.black color={theme.text2} fontWeight={500} fontSize={14}>
+                Recipient
+              </TYPE.black>
+              {data.address && (
+                <Link href={getEtherscanLink(chainId, data.address, 'address')} style={{ fontSize: '14px' }}>
+                  (View on Etherscan)
+                </Link>
+              )}
+              {data.name && (
+                <Link href={getEtherscanLink(chainId, data.name, 'address')} style={{ fontSize: '14px' }}>
+                  (View on Etherscan)
+                </Link>
+              )}
```

Comment:
shouldn't this be `data.name ? ... : ...`? i believe `data.address` and `data.name` can both be defined, which would lead to doubling